orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationClient.java

```diff
@@ -57,27 +59,32 @@ public class OrchestrationClient {
             .build();
   }
 
-  @Nonnull private final Supplier<AiCoreDeployment> deployment;
+  @Nonnull private final Supplier<HttpDestination> destinationSupplier;
 
   /** Default constructor. */
   public OrchestrationClient() {
-    deployment = () -> new AiCoreService().forDeploymentByScenario("orchestration");
+    destinationSupplier =
+        () ->
+            new AiCoreService()
+                .getDestinationForDeploymentByScenario(
+                    AiCoreService.DEFAULT_RESOURCE_GROUP, DEFAULT_SCENARIO);
   }
 
   /**
-   * Constructor with a custom deployment, allowing for a custom resource group or otherwise
-   * specific deployment ID.
+   * Constructor with a custom destination, allowing for a custom resource group or otherwise custom
+   * destination.
    *
    * <p>Example:
    *
    * <pre>{@code
-   * new OrchestrationClient(new AiCoreService().forDeploymentByScenario("orchestration"));
+   * new OrchestrationClient(new AiCoreService().getDestinationForDeploymentByScenario("custom-rg", "orchestration"));
    * }</pre>
    *
-   * @param deployment The specific {@link AiCoreDeployment} to use.
+   * @param destination The specific {@link HttpDestination} to use.
+   * @see AiCoreService#getDestinationForDeploymentByScenario(String, String)
    */
-  public OrchestrationClient(@Nonnull final AiCoreDeployment deployment) {
```

Comment:
(Question/Minor)

I think after the change you need to document how to interact with orchestration service with `resource-group <> "default"`.

Reply:
The level of documentation doesn't really change before / after this PR I think..? But yes, should be e2e tested and documented as part of the [related ticket](https://github.tools.sap/AI/ai-sdk-java-backlog/issues/123), thus I would leave that out of this PR.

Reply:
I just realized, you guys botched immutability for `withResourceGroup(...)` in #100.

I guess, that's on me for not adding unit tests for every immutability contract ._.

Reply:
> The level of documentation doesn't really change before / after this PR I think..?

To me this is surprising too... I assumed we had docs on that. We can discuss in Slack. I agree, it's not a requirement for your PR.